### ISSUE TITLE
False positive for W0631

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#178]
Originally reported by: **the mulhern (BitBucket: [the_mulhern](http://bitbucket.org/the_mulhern))**

----------------------------------------

Consider the following method:

```
#!python

def method():
    for var1 in <iterable>:
        if var1 == <some value>:
            break
    else:
        return

    print var1
```

W0631 will report "Using possibly undefined variable 'var1'." But, I think it should be possible to detect that, if <iterable> is empty the else clause is taken and the function returns. Therefore, at the line where var1 is used, "print var1", var1 must be defined.

This is pretty minor, especially as it might require more heavyweight analysis than W0631 is set up for at this time, but I don't know how that analysis works, so I just point it out here.

----------------------------------------
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/178

**Comment by PCManticore** [comment]
Closing, I think this would require a too heavy weight analysis than what the check should require, especially since it feels an edge case.
